### DISCUSSION TITLE
How to update related entities using Entity Framework Core

### DISCUSSION THREAD

**Discussion by karamem0** [Avanade/Beef#197] [Q&A] [answered]
I am using Beef with Entity Framework Core. I want to create an API that updates related entities like `Order` and `OrderDetails`. Pure Entity Framework Core can track child entities when `Add` or `Remove` method of relationship property was called, but Beef seems not to be able to track child entities. I think this is caused by AutoMapper mapping `Entities` to `EfModel`.
Any good ideas on this?

**Comment by chullybun** [comment, marked as answer]
Are you using the latest version of Beef; i.e. v5 - see previous comments in [Issue 94](https://github.com/Avanade/Beef/issues/194). Extra effort / changes were made to support more advanced EF scenarios, such as what you describe. There is a new [sample](https://github.com/Avanade/Beef/tree/version-5/samples/MyEf.Hr) (including documentation) that demonstrates a parent to many child relationship. There may be scenarios that [`CoreEx`](https://github.com/Avanade/CoreEx/tree/main/src/CoreEx.EntityFrameworkCore)/`Beef` do not cover; in these instances then standard EF should be used. Latest is also moving away from AutoMapper as difficult to debug, etc.
